base/arraymath.jl

```diff
@@ -35,26 +35,19 @@ function !(A::AbstractArray{Bool})
 end
 
 ## Binary arithmetic operators ##
-
-promote_array_type{Scalar, Arry}(F, ::Type{Scalar}, ::Type{Arry}) = promote_op(F, Scalar, Arry)
-promote_array_type{S<:Real, A<:AbstractFloat}(F, ::Type{S}, ::Type{A}) = A
-promote_array_type{S<:Integer, A<:Integer}(F, ::Type{S}, ::Type{A}) = A
-promote_array_type{S<:Integer}(F, ::Type{S}, ::Type{Bool}) = S
-promote_array_type(F, ::Type{Bool}, ::Type{Bool}) = promote_op(F, Bool, Bool)
-
-# Handle operations that return different types
-./(x::Number, Y::AbstractArray) =
-    reshape([ x ./ y for y in Y ], size(Y))
-./(X::AbstractArray, y::Number) =
-    reshape([ x ./ y for x in X ], size(X))
-.\(x::Number, Y::AbstractArray) =
-    reshape([ x .\ y for y in Y ], size(Y))
-.\(X::AbstractArray, y::Number) =
-    reshape([ x .\ y for x in X ], size(X))
-.^(x::Number, Y::AbstractArray) =
-    reshape([ x ^ y for y in Y ], size(Y))
-.^(X::AbstractArray, y::Number      ) =
-    reshape([ x ^ y for x in X ], size(X))
+promote_array_type{S<:Number, A<:AbstractArray}(F, ::Type{S}, ::Type{A}) =
```

Comment:
Too bad for `@pure`, let's hope he doesn't mind emails from Github. (next time use `` ` `` to wrap macros, because otherwise github pings and subscribes users named the same as our macros)